comptime: struct fixed array field getting value causes C error

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 d3ed361, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 c1d1efb2091e989dbb09e38b308c82c2cade1aa8.d3ed361
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 22.04.5 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-4150 CPU @ 3.50GHz
|Memory              |1.19GB/3.7GB
|                    |
|V executable        |/home/jorge/v/v
|V last modified time|2025-11-18 17:05:55
|                    |
|V home dir          |OK, value: /home/jorge/v
|VMODULES            |OK, value: /home/jorge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/jorge/bugs
|                    |
|Git version         |git version 2.34.1
|V git status        |weekly.2025.16-1084-gd3ed3614-dirty
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
|gcc version         |gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
|clang version       |Ubuntu clang version 14.0.0-1ubuntu1.1
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.5 ()
|glibc version       |ldd (Ubuntu GLIBC 2.35-0ubuntu3.11) 2.35

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  251118.v && 251118`
```v
struct FutureUnion {
	number u64
	bytes  [8]u8 //@[skip]
}

fn encode_union[T](t T) {
	$for f in T.fields {
		v := t.$(f.name)
		dump('${f.name} ${v}')
	}
}

fn main() {
	encode_union(FutureUnion{
		number: 256
		bytes: [u8(0),0,0,0,0,0,0,0]! // init here didn't help
	})
}


```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/251118.01KABYWNSRD8M9WDKYJ1Z581EJ.tmp.c:5415: error: '{' expected (got ";")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

print name and value of struct fixed array field

**Details**

Related with issue #25607 . Here the expected `union` still is a struct and don't using `json2` yet. Simple generic encoder code can't get the value `t.$(f.name)` of fixed array field even when explicitly initialized:
```
/tmp/v_1000/251118.01KABYV610YDKN3ZDAB47FSP7S.tmp.c: In function ‘main__encode_union_T_main__FutureUnion’:
/tmp/v_1000/251118.01KABYV610YDKN3ZDAB47FSP7S.tmp.c:5415:46: error: invalid initializer
 5415 |                         Array_fixed_u8_8 v = t.bytes;
      |                                              ^
```
Maybe reading attribute `@[skip]` could prevent needing reading any fixed array value, but it should be read in any other cases.

Thanks in advance V team!
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -532,7 +532,7 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 		unaliased_right_sym := g.table.final_sym(unwrapped_val_type)
 		unaliased_left_sym := g.table.final_sym(g.unwrap_generic(var_type))
 		is_fixed_array_var := unaliased_right_sym.kind == .array_fixed && val !is ast.ArrayInit
-			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr, ast.InfixExpr]
+			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.ComptimeSelector, ast.DumpExpr, ast.InfixExpr]
 			|| (val is ast.CastExpr && val.expr !is ast.ArrayInit)
 			|| (val is ast.PrefixExpr && val.op == .arrow)
 			|| (val is ast.UnsafeExpr && val.expr in [ast.SelectorExpr, ast.Ident, ast.CallExpr]))
@@ -568,7 +568,8 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 				} else {
 					g.write('{${styp} _ = ')
 				}
-				if val in [ast.MatchExpr, ast.IfExpr] && unaliased_right_sym.info is ast.ArrayFixed {
+				if (val in [ast.MatchExpr, ast.IfExpr, ast.ComptimeSelector] || is_fixed_array_var)
+					&& unaliased_right_sym.info is ast.ArrayFixed {
 					tmp_var := g.expr_with_var(val, var_type, false)
 					g.fixed_array_var_init(tmp_var, false, unaliased_right_sym.info.elem_type,
 						unaliased_right_sym.info.size)

--- a/vlib/v/tests/comptime/comptime_for_fixed_array_field_test.v
+++ b/vlib/v/tests/comptime/comptime_for_fixed_array_field_test.v
@@ -1,0 +1,86 @@
+struct StructWithFixedArrays {
+	number u64
+	bytes  [8]u8
+	floats [4]f32
+	name   string
+	matrix [2][3]int
+}
+
+fn collect_field_info[T](t T) []string {
+	mut info := []string{}
+	$for f in T.fields {
+		v := t.$(f.name)
+		info << '${f.name}: ${v}'
+	}
+	return info
+}
+
+fn count_accessible_fields[T](t T) int {
+	mut count := 0
+	$for f in T.fields {
+		_ := t.$(f.name)
+		count++
+	}
+	return count
+}
+
+fn test_fixed_array_field_access() {
+	test := StructWithFixedArrays{
+		number: 42
+		bytes:  [u8(1), 2, 3, 4, 5, 6, 7, 8]!
+		floats: [f32(1.0), 2.0, 3.0, 4.0]!
+		name:   'test'
+		matrix: [[1, 2, 3]!, [4, 5, 6]!]!
+	}
+
+	info := collect_field_info(test)
+
+	assert info.len == 5
+	assert info[0].contains('number: 42')
+	assert info[1].contains('bytes: [1, 2, 3, 4, 5, 6, 7, 8]')
+	assert info[2].contains('floats: [1.0, 2.0, 3.0, 4.0]')
+	assert info[3].contains('name: test')
+	assert info[4].contains('matrix: [[1, 2, 3], [4, 5, 6]]')
+}
+
+fn test_nested_fixed_arrays() {
+	test := StructWithFixedArrays{
+		number: 100
+		bytes:  [u8(10), 20, 30, 40, 50, 60, 70, 80]!
+		floats: [f32(0.1), 0.2, 0.3, 0.4]!
+		name:   'nested'
+		matrix: [[10, 20, 30]!, [40, 50, 60]!]!
+	}
+
+	assert count_accessible_fields(test) == 5
+}
+
+fn test_fixed_array_comptime_iteration() {
+	test := StructWithFixedArrays{
+		number: 999
+		bytes:  [u8(0), 1, 2, 3, 4, 5, 6, 7]!
+		floats: [f32(1.1), 2.2, 3.3, 4.4]!
+		name:   'iteration'
+		matrix: [[100, 200, 300]!, [400, 500, 600]!]!
+	}
+
+	mut field_names := []string{}
+	mut field_values := []string{}
+
+	$for f in StructWithFixedArrays.fields {
+		field_names << f.name
+		val := test.$(f.name)
+		field_values << '${val}'
+	}
+	assert field_names.len == 5
+	assert field_values.len == 5
+
+	assert 'number' in field_names
+	assert 'bytes' in field_names
+	assert 'floats' in field_names
+	assert 'name' in field_names
+	assert 'matrix' in field_names
+
+	assert field_values[0].contains('999')
+	assert field_values[3].contains('iteration')
+}

